src/components/pages/PageDecksEdit/PageDecksEdit.jsx

```diff
@@ -0,0 +1,237 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { If, Else } from 'react-if';
+import { useHistory, useParams } from 'react-router';
+import { useSelector } from 'react-redux';
+import MultiSelect from 'react-multi-select-component';
+import TemplateDefault from '../../Templates/TemplateDefault';
+import LoadingAnim from '../../atoms/LoadingSpinner/LoadingSpinner';
+import firebaseCollectionsHelper from '../../../firebase/firebase-collections-helper/firebase-collections-helper';
+
+function PageDecksEdit() {
+  const { id } = useParams();
+  const user = useSelector((state) => state.user);
+  const history = useHistory();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [characterOptions, setCharacterOptions] = useState([]);
+  const [selectedCharacterIds, setSelectedCharacterIds] = useState([]);
+  const [circumstanceOptions, setCircumstanceOptions] = useState([]);
+  const [selectedCircumstanceIds, setSelectedCircumstanceIds] = useState([]);
+  const [conflictOptions, setConflictOptions] = useState([]);
+  const [selectedConflictIds, setSelectedConflictIds] = useState([]);
+  const [deck, setDeck] = useState({});
+
+  const getCardOptions = useCallback(
+    async (cardType) => {
+      const conflictCards = await firebaseCollectionsHelper
+        .getAllCardsDataByType(user.uid, cardType);
+
+      return conflictCards.map((card) => ({
+        label: card.cardTitle,
+        value: card.id,
+      }));
+    }, [user.uid],
+  );
+
+  const populateCharacters = useCallback(
```

Comment:
All of your populate methods can be combined into a single function `populate(options, setter)`.